src/services/walletconnect/WalletConnectContext.tsx

```diff
@@ -25,7 +28,8 @@ export const WalletConnectProvider = ({ children }: { children: ReactNode }) =>
     safeAddress,
   } = useSafeInfo()
   const [walletConnect, setWalletConnect] = useState<WalletConnectWallet | null>(null)
-  const [wcUri, setWcUri] = useWalletConnectSearchParamUri()
+  const [searchParamWcUri, setSearchParamWcUri] = useWalletConnectSearchParamUri()
+  const [clipboardWcUri, setClipboardWcUri] = useWalletConnectClipboardUri()
```

Comment:
I wouldn't overload the context with this.

I would listen to the clipboard/URL from the header widget, and open the poup with the input if a wc-URI is detected. Then populate the input with that, and trigger a propose as usual from the input.